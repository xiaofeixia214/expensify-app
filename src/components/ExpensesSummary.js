import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selecctExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral(expensesTotal / 1000).format('$0,0.00');
	return (
		<div>
			<h1>View {expenseCount} {expenseWord} totaling {formattedExpensesTotal}</h1>
		</div>
	);
};

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: selecctExpensesTotal(visibleExpenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);