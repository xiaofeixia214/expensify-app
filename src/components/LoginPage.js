import React from 'react';
import { connect } from 'react-redux';
import { startLoginWithGoogle, startLoginWithGithub } from '../actions/auth';

export const LoginPage = ({ startLoginWithGoogle, startLoginWithGithub }) => (
	<div className="box-layout">
		<div className="box-layout__box">
			<h1 className="box-layout__title">Expensify</h1>
			<p>It's time to get your expenses under control.</p>
			<button className="button box-layout__button" onClick={startLoginWithGoogle}>Login with Goolge</button>
			<button className="button" onClick={startLoginWithGithub}>Login with GitHub</button>
		</div>
	</div>
);

const mapDispatchToProps = (dispatch) => ({
	startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
	startLoginWithGithub: () => dispatch(startLoginWithGithub())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);