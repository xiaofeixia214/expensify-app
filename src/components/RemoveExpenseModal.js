import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (
	<Modal
		isOpen={props.showRemoveExpenseModal}
		contentLabel="Confirmation"
		closeTimeoutMS={100}
		className="modal"
	>
		<p className="modal__body">Do you want to delete this expense?</p>
		<div className="modal__button-bar">
			<button className="button" onClick={props.handleRemoveExpense}>Yes</button>
			<button className="button" onClick={props.hanldeCancelRemoveExpense}>No</button>
		</div>
	</Modal>
);

export default RemoveExpenseModal;