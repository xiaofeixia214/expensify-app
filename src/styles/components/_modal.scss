.ReactModalPortal > div {
	opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
	align-items: center;
	display: flex;
	justify-content: center;
	transition: opacity 200ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
	opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
	opacity: 0;
}

.ReactModal__Body--open {
	overflow-y: hidden;
}

.modal {
	background: fade-out($dark-blue, .5);
	color: white;
	max-width: 40rem;
	outline: none;
	padding: $l-size;
	text-align: center;
}

.modal__body {
	font-size: 2rem;
	font-weight: 300;
}

.modal__button-bar {
	display: flex;
	justify-content: center;
	button {
		margin: 0 $m-size;
	}
}