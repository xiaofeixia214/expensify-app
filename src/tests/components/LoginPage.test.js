import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage correctly', () => {
	const wrapper = shallow(<LoginPage />);
	expect(wrapper).toMatchSnapshot();
});

test('should call startLoginWithGoogle on google button click', () => {
	const startLoginWithGoogle = jest.fn();
	const wrapper = shallow(<LoginPage startLoginWithGoogle={startLoginWithGoogle} />);
	wrapper.find('button').first().simulate('click');
	expect(startLoginWithGoogle).toHaveBeenCalled();
});

test('should call startLoginWithGithub on google button click', () => {
	const startLoginWithGithub = jest.fn();
	const wrapper = shallow(<LoginPage startLoginWithGithub={startLoginWithGithub} />);
	wrapper.find('button').at(1).simulate('click');
	expect(startLoginWithGithub).toHaveBeenCalled();
});