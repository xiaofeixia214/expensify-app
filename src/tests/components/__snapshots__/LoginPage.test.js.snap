// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <button
      className="button box-layout__button"
    >
      Login with Goolge
    </button>
    <button
      className="button"
    >
      Login with GitHub
    </button>
  </div>
</div>
`;
